src/index.ts

```diff
@@ -112,7 +114,40 @@ export async function parseSwap({
           amount: nativeAmountToTaker,
           address: NATIVE_TOKEN_ADDRESS,
         };
+  if (to?.toLowerCase() === FORWARDING_MULTICALL_ADDRESS.toLowerCase()) {
+    const { args: multicallArgs } = decodeFunctionData({
+      abi: FORWARDING_MULTICALL_ABI,
+      data: transaction.input,
+    });
 
+    if (multicallArgs && Array.isArray(multicallArgs) && multicallArgs[0] && Array.isArray(multicallArgs[0])) {
+      const { args: settlerArgs } = decodeFunctionData({
+        abi: SETTLER_META_TXN_ABI,
+        data: multicallArgs[0][1]?.data,
+      });
+
+      const recipient =
+        settlerArgs[0].recipient.toLowerCase() as Address;
+
+      const msgSender = settlerArgs[3];
+
+      const nativeAmountToTaker = calculateNativeTransfer(trace, {
+        recipient,
+      });
+
+      if (nativeAmountToTaker === "0") {
+        [output] = logs.filter(
+          (log) => log.to.toLowerCase() === msgSender.toLowerCase()
+        );
+      } else {
+        output = {
+          symbol: NATIVE_SYMBOL_BY_CHAIN_ID[chainId],
+          amount: nativeAmountToTaker,
+          address: NATIVE_TOKEN_ADDRESS,
+        };
+      }
+    }
+  }
```

Comment:
MultiCall3 will no longer be used and we can instead rely on the forwarding multi call address and abi @ this step of txn parsing. I decided to leave the previous logic related to Multicall3 to maintain backwards compatibility.
cc: @duncancmt